crates/shielded_token/src/storage.rs

```diff
@@ -20,13 +22,17 @@ where
         max_reward_rate: max_rate,
         kd_gain_nom,
         kp_gain_nom,
-        locked_ratio_target: locked_target,
+        locked_amount_target,
     } = params;
     storage.write(&masp_last_inflation_key(address), Amount::zero())?;
-    storage.write(&masp_last_locked_ratio_key(address), Dec::zero())?;
+    storage.write(&masp_last_locked_amount_key(address), Amount::zero())?;
     storage.write(&masp_max_reward_rate_key(address), max_rate)?;
-    storage.write(&masp_locked_ratio_target_key(address), locked_target)?;
     storage.write(&masp_kp_gain_key(address), kp_gain_nom)?;
     storage.write(&masp_kd_gain_key(address), kd_gain_nom)?;
+
+    let raw_target = Uint::from(*locked_amount_target)
```

Comment:
`locked_amount_target` should probably be a `Uint` or `Amount`. If that is the case, then this multiplication (or any other scaling operation) should not be needed.